### PR TITLE
Style Standard Elements

### LINKED ISSUE #734: Don't style form inputs on Firefox
Setting `border-color` or `background-color` on any `button` or `input` causes them to stop looking like native components and instead makes them look like some ancient OS.

What buttons look like in Firefox when styled by Dark Reader: oh god

What they should look like: please

Chrome ignores this, but Firefox makes a non-native button instead. Not styling form inputs, or having an option for it, would help make the browser look more integrated with the OS.

### LINKED ISSUE #3239: [BUG] Text boxes and drop-down menus look like Win98 style when addon enabled
### **Describe the bug**

When the dark reader add-on is enabled, drop-down menus and text boxes that have not been stylized by the website change from their normal, rounded look that is the OS default, to squared ones that look to be from Windows 98. See screenshots, the website I used as an example was the AUR page for anaconda.

Add-on disabled:
Screenshot_20200726_104031

Add-on enabled:
Screenshot_20200726_104048

You can see how the Keywords text box and the Search By drop-down are changed to look different from how they do originally.

### **Steps to reproduce**

- Go to any website with unstylized drop-down menus or text boxes (I used [AUR's page on anaconda](https://aur.archlinux.org/packages/anaconda/) as an example)
- Enable/Disable Dark reader and look at the difference in the boxes

### **Expected behavior**

The drop-down menus and text boxes should stay the same shape/style, just darker.

### **Actual behavior**

The drop-down menus and text boxes are replaced with differently shaped/styled ones that look outdated.

### **Screenshots**

Screenshots are in the first section.

### **System Information:**
 - OS: Arch Linux with KDE Plasma 5.19.3
 - Browser: Firefox Nightly 80.0a1 (2020-07-25) (though I can confirm the bug exists in Firefox Stable, too.)
 - Dark Reader version: 4.9.15

### PR DESCRIPTION
**Pull request by Gusted**
- Add option to style standard elements e.g. button, input, select.
- Resolves #734
- Resolves #3239

### CODE REVIEW

File: src/inject/dynamic-theme/modify-css.ts [hunk 1]
```diff
@@ -51,40 +51,52 @@ export function getModifiableCSSDeclaration(property: string, value: string, rul
...
-    lines.push('input:-webkit-autofill,');
-    lines.push('textarea:-webkit-autofill,');
-    lines.push('select:-webkit-autofill {');
-    lines.push(`    background-color: ${modifyBackgroundColor({r: 250, g: 255, b: 189}, filter)} !important;`);
-    lines.push(`    color: ${modifyForegroundColor({r: 0, g: 0, b: 0}, filter)} !important;`);
-    lines.push('}');
-    if (filter.scrollbarColor) {
-        lines.push(getModifiedScrollbarStyle(filter));
+    if (styleStandardElements) {
+        lines.push(`${isIFrame ? '' : 'html, body, '}input, textarea, select, button {`);
```
Comment (alexanderby) [inline src/inject/dynamic-theme/modify-css.ts:62]: Try modifying selectors only, push `input, select, button` when necessary.

Comment (Gusted) [reply src/inject/dynamic-theme/modify-css.ts:62]:
```CSS 
 {
    background-color: green;
}
```
Wow an empty css rule in a IFrame

File: src/definitions.d.ts [hunk 2]
```diff
@@ -54,6 +54,7 @@ export interface Theme {
     lightSchemeTextColor: string;
     scrollbarColor: '' | 'auto' | string;
     selectionColor: '' | 'auto' | string;
+    styleStandardElements: boolean;
```
Comment (alexanderby) [inline src/definitions.d.ts:57]: Maybe it's worth renaming to `styleFormControls` (understood by web devs) or `styleSystemControls` (maybe more clear for users)? Maybe you will have some other ideas.

File: src/inject/dynamic-theme/modify-css.ts [hunk 3]
```diff
@@ -51,40 +51,40 @@ export function getModifiableCSSDeclaration(property: string, value: string, rul
...
     const lines: string[] = [];
     if (!isIFrame) {
         lines.push('html {');
-        lines.push(`    background-color: ${modifyBackgroundColor({r: 255, g: 255, b: 255}, filter)} !important;`);
+        lines.push(`    background-color: ${modifyBackgroundColor({r: 255, g: 255, b: 255}, theme)} !important;`);
         lines.push('}');
     }
-    lines.push(`${isIFrame ? '' : 'html, body, '}input, textarea, select, button {`);
-    lines.push(`    background-color: ${modifyBackgroundColor({r: 255, g: 255, b: 255}, filter)};`);
+    lines.push(`${isIFrame ? '' : 'html, body, '}${styleStandardElements ? '' : 'input, textarea, select, button'} {`);
```
Comment (alexanderby) [inline src/inject/dynamic-theme/modify-css.ts:61]: So, when the option is `true`, should it apply style or not? It doesn't in your case.

File: src/inject/dynamic-theme/modify-css.ts [hunk 4]
```diff
@@ -51,40 +51,40 @@ export function getModifiableCSSDeclaration(property: string, value: string, rul
...
+    lines.push(`    background-color: ${modifyBackgroundColor({r: 255, g: 255, b: 255}, theme)};`);
     lines.push('}');
     lines.push('html, body, input, textarea, select, button {');
```
Comment (alexanderby) [inline src/inject/dynamic-theme/modify-css.ts:64]: Not sure if it should do the same here.

File: src/inject/dynamic-theme/modify-css.ts (hunk 3 above)
Comment (Gusted) [reply src/inject/dynamic-theme/modify-css.ts:61]: 🤦🏽‍♀️

File: src/defaults.ts [hunk 5]
```diff
@@ -30,6 +30,7 @@ export const DEFAULT_THEME: Theme = {
     lightSchemeTextColor: DEFAULT_COLORS.lightScheme.text,
     scrollbarColor: isMacOS() ? '' : 'auto',
     selectionColor: 'auto',
+    styleStandardElements: true,
```
Comment (alexanderby) [inline src/defaults.ts:33]: Old name here, use `styleSystemControls`

File: src/ui/connect/mock.ts [hunk 6]
```diff
@@ -19,6 +19,7 @@ export function getMockData(override = {} as Partial<ExtensionData>): ExtensionD
                 engine: 'cssFilter',
                 stylesheet: '',
                 scrollbarColor: 'auto',
+                styleStandardElements: false,
```
Comment (alexanderby) [inline src/ui/connect/mock.ts:22]: Old name here. Also make it `true`

File: src/ui/popup/theme/controls/style-standard-elements.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import {m} from 'malevic';
+import {DropDown} from '../../../controls';
+import ThemeControl from './theme-control';
+
+export default function StyleStandardElements(props: {value: boolean; onChange: (boolean: any) => void}) {
+    const options = [{id: true, content: 'Yes'}, {id: false, content: 'No'}];
+    return (
+        <ThemeControl label="Style Standard Elements">
```
Comment (alexanderby) [inline src/ui/popup/theme/controls/style-standard-elements.tsx:8]: *Style system controls*

File: src/ui/popup/theme/controls/index.tsx [hunk 8]
```diff
@@ -9,6 +9,7 @@ import Scheme from './scheme';
 import Scrollbar from './scrollbar';
 import SelectionColorEditor from './selection';
 import Sepia from './sepia';
+import StyleStandardElements from './style-standard-elements';
```
Comment (alexanderby) [inline src/ui/popup/theme/controls/index.tsx:12]: Let's make a file name and a component function name to correspond with the setting name.

**Review by alexanderby — CHANGES_REQUESTED** [review]

**Review by alexanderby — APPROVED** [review]

**Pull request merged**